### DISCUSSION TITLE
introduce context usage in the service client file.

### DISCUSSION THREAD

**Discussion by pkenchap** [gophercloud/gophercloud#2671] [Ideas]
Context usage in the gopher cloud gives more request handling and goroutines working on a request typically needs access to request-specific values such as the identity of the end user, authorization tokens, and the request’s deadline. if we enable the context which provide easy integration external tools .

why we are proposing to have a context in the client is to integration of **Opentelemetry (**https://github.com/open-telemetry/opentelemetry-go) in the Gophercloud sdk (https://github.com/gophercloud/gophercloud) . provides the tracing of the API can be monitored with the timelines, which gives immense use for the performance test for calculation individual api time taken for the operation. Also gives a nice UI based graphs using the JaegerUI .

**Comment by pierreprinetti** [comment]
dup of https://github.com/gophercloud/gophercloud/issues/2670
